### PR TITLE
GH-102341: Improve the test function for pow

### PR DESCRIPTION
**Pull request by ppmfloss**
Currently in the powtest function in Lib/test/test_pow.py file we are not testing the results of calling pow with negative exponents.
This PR will add appropriate assert statements to test the results of calling pow with negative exponents.

* Issue: gh-102341

### CODE REVIEW

**Comment by terryjreedy** [comment]
Mark and Raymond, a one-line addition to test_pow that looks plausible but I am not sure.

Pipelines win32 failure is unrelated of type seen before.

**Comment by mdickinson** [comment]
Thanks for the ping, @terryjreedy. I'll take a look this weekend.

**Review by mdickinson — APPROVED** [review]
LGTM. Can we drop the `for othertype in (int,):` outer loop while we're here? (I'm guessing that it stems from Python 2 days, and was originally spelled `for othertype in (int, long):`.)

File: Lib/test/test_pow.py [hunk 1]
```diff
@@ -22,9 +22,10 @@ def powtest(self, type):
             for othertype in (int,):
                 for i in list(range(-10, 0)) + list(range(1, 10)):
                     ii = type(i)
+                    inv = pow(ii, -1) # inverse of ii
                     for j in range(1, 11):
                         jj = -othertype(j)
```
Comment (terryjreedy) [inline Lib/test/test_pow.py]:
```suggestion
                for jj in range(-10, 0):
```

File: Lib/test/test_pow.py [hunk 2]
```diff
@@ -22,9 +22,10 @@ def powtest(self, type):
             for othertype in (int,):
                 for i in list(range(-10, 0)) + list(range(1, 10)):
                     ii = type(i)
+                    inv = pow(ii, -1) # inverse of ii
```
Comment (terryjreedy) [inline Lib/test/test_pow.py]:
```suggestion
            for i in list(range(-10, 0)) + list(range(1, 10)):
                ii = type(i)
                inv = pow(ii, -1) # inverse of ii
```

File: Lib/test/test_pow.py [hunk 3]
```diff
@@ -22,9 +22,10 @@ def powtest(self, type):
             for othertype in (int,):
```
Comment (terryjreedy) [inline Lib/test/test_pow.py]: Mark is correct: (int,) used to be (int, float). If this line is deleted, other changes below must be applied.  For some reason, I could not make them all at once.
```suggestion
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]

**Comment by terryjreedy** [comment]
Please recheck after deletion, dedent, and condensation.

**Comment by ppmfloss** [comment]
I have removed some indentation spaces and retested it locally successfully.

**Review by mdickinson — APPROVED** [review]
LGTM!

**Pull request merged**